### PR TITLE
Carrier Eggsac Strain can now place eggs on Regular Weeds, Egg range Bugfix, and QOL notification for Egg generation

### LINKED ISSUE #3869: Eggsac Carrier/Ovi Queen Egg Planting Range
### Testmerges

N/A

### Description of the bug

The Eggsac Carrier's tooltip explains that they are able to plant eggs an additional tile away, **but they aren't**. In the code, they have an increased building range - but that building range has no input on how far away the can plant the eggs. This most likely has something to do with the fact that xenomorph eggs check for proximity. Ovipositor Queen also suffers from the same bug, as it was intended for her to be able to plant eggs up to 3 tiles away - which she is not capable of.

grafik
grafik
grafik
grafik

### What's the difference with what should have happened?

Currently: Eggsac Carrier and Ovipositor Queen have the regular planting range of 1 tile.
Expected: Eggsac Carrier should have planting range of 2 tiles, Ovipositor Queen should have planting range of 3 tiles.

### How do we reproduce this bug?

Any attempt to plant an egg further than a tile away makes the bug self-apparent.

### Issue Bingo

- [X] Issue could be reproduced at least once
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

### PR DESCRIPTION
**Pull request by Ben10083**
# About the pull request

Eggsac Strain can now place eggs on regular weeds instead of only hive weeds, notified when they generate an egg, and fixes #3869 

# Explain why it's good for the game

The eggsac strain is designed to be a mobile egg factory, able to place eggs closer to front for player huggers/general captures. However with limitations on only hiveweeds, it really limited the ability of the eggsac strain to utilize their strain's abilities, and the risk of losing a egg to marines forced them to stay at hive, making it a unpopular pick. This gives more purpose to the eggsac, allowing them to place eggs much closer to frontline for ghosts to spawn into/general captures, as well as an improv trap for approaching marines. They still have the risk of giving eggs to marines however, which I feel balances out this buff. Also bugs bad

# Testing Photographs and Procedure
Things I have tested:
<details>

- Carrier no strain range at 1
- Carrier Eggsac strain range at 3
- Drone has appropriate egg range
- Queen ovi has longer egg plant range
- Queen deovi back to basic 1 egg plant range

</details>

# Changelog
:cl:
balance: Carriers with Eggsac strain can now place eggs on regular weeds.
fix: Extra egg planting range now properly applied for Eggsac strain and Queen on ovi
qol: Eggsac Carriers now notified when they generate a egg.
code: new variable to xenomorphs to dictate egg planting distance, and removal of unused procs
/:cl:

### CODE REVIEW

**Comment by Diegoflores31** [comment]
while looking at this any chance the egg generation cooldown is reduced a little ? its comically slow currently

**Comment by Ben10083** [comment]
> while looking at this any chance the egg generation cooldown is reduced a little ? its comically slow currently

I am sure it can be done, but I don't want to expand the scope of this pr anymore

File: code/modules/mob/living/carbon/xenomorph/egg_item.dm [hunk 1]
```diff
@@ -82,13 +85,20 @@
 		return
 
 	var/obj/effect/alien/weeds/hive_weeds = null
-	for(var/obj/effect/alien/weeds/W in T)
-		if(W.weed_strength >= WEED_LEVEL_HIVE && W.linked_hive.hivenumber == hivenumber)
-			hive_weeds = W
+	var/obj/effect/alien/weeds/any_weeds = null
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/egg_item.dm:88]: = null is redundant

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 2]
```diff
@@ -281,6 +281,7 @@
 	// Related to zooming out (primarily queen and boiler)
 	var/devour_timer = 0 // The world.time at which we will regurgitate our currently-vored victim
 	var/extra_build_dist = 0 // For drones/hivelords. Extends the maximum build range they have
+	var/egg_planting_range = 1 //tiles from self you can plant eggs.
```
[the remark this answers is not in the document]
Comment (Ben10083) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm]: Resolved, also had the others near it changed

**Review by harryob — APPROVED** [review]
code looks fine just resolve ur conflicts

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Pull request merged**